### DISCUSSION TITLE
Nix flake configuration documentation

### DISCUSSION THREAD

**Discussion by RossSmyth** [helix-editor/helix#11528] [Q&A]
Hello. On Windows I build Helix from source on nightly with the `unicode-lines` feature enabled.

I would like to replicate this on Linux with Nix. I have it setup so that I can compile it from source, but just the default config. Which is with stable rustc (I think, maybe MSRV my nix isn't very good), and without any features enabled. 

I would like to request whoever maintains the flake to add some documentation to it, as it is a bit dense and not very clear on how to override things, or if you even can. My attempts so far have failed.

**Comment by the-mikedavis** [comment]
I sort of maintain the flake / Nix stuff since rewriting it in https://github.com/helix-editor/helix/pull/7763. I'm not sure how the overlay stuff is meant to be used either though (I use `nix profile install/upgrade`) or if it's possible to specify feature flags - you may be better off pinging those who introduced it to ask: https://github.com/helix-editor/helix/pull/7078, https://github.com/helix-editor/helix/pull/8749.

**Comment by RossSmyth** [comment]
Ok, @happysalada or @mydumpfire

I have a few question about the Nix setup. I am relatively new to Nix, so bear with me. Though my friend who works with Nix full time said this flake confused him as well. Basically there are two things I would like to do so that it is equivalent to my Windows dev setup:

1. Compile Helix with Rust Nightly

I do this as I help with rustc work and Helix is a pretty hefty codebase that can and has exposed some edge cases. Also possible slightly better codegen.

2. Compile Helix with the "unicode-lines" feature

**Reply by RossSmyth** [reply]
For reference you can find the relevant lines of my config here:
https://github.com/RossSmyth/nixos/blob/f96b8633f03c39be56f60ff4ac707b3f3fab7b3e/flake.nix#L10
https://github.com/RossSmyth/nixos/blob/f96b8633f03c39be56f60ff4ac707b3f3fab7b3e/home-manager/home.nix#L8-L16
https://github.com/RossSmyth/nixos/blob/f96b8633f03c39be56f60ff4ac707b3f3fab7b3e/home-manager/helix.nix
